enumerable_methods.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    each do |element|
+      yield(element, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    array = []
+    my_each do |i|
+      array << i if yield(i)
+    end
+    array
+  end
+
+  def my_all?
+    obj = self
+    obj.my_each do |i|
+      return false unless yield i
+      end
+      true
+    end
```

Comment:
Please delete this `end`  :point_up: 
No need of `end` when using `guard clause`.
```suggestion
```